tests/acceptance/smoke-test-slow.js

```diff
@@ -51,263 +47,225 @@ describe('Acceptance: smoke-test', function() {
-      let appJsPath = path.join(appRoot, rootPath, 'app.js');
+  it('ember new foo, build development, and verify generated files', async function() {
+    await runCommand(path.join('.', 'node_modules', 'ember-cli', 'bin', 'ember'), 'build');
 
-      let result = yield runCommand(path.join('.', 'node_modules', 'ember-cli', 'bin', 'ember'), 'build');
-      expect(result.code).to.equal(0, `expected exit code to be zero, but got ${result.code}`);
+    let dirPath = path.join(appRoot, 'dist');
+    let paths = walkSync(dirPath);
 
-      // add something broken to the project to make build fail
-      fs.appendFileSync(appJsPath, '{(syntaxError>$@}{');
+    expect(paths).to.have.length.below(24, `expected fewer than 24 files in dist, found ${paths.length}`);
+  });
 
-      result = yield expect(runCommand(path.join('.', 'node_modules', 'ember-cli', 'bin', 'ember'), 'build')).to.be
-        .rejected;
+  it('ember build exits with non-zero code when build fails', async function() {
+    let rootPath = isExperimentEnabled('MODULE_UNIFICATION') ? 'src' : 'app';
+    let appJsPath = path.join(appRoot, rootPath, 'app.js');
 
-      expect(result.code).to.not.equal(0, `expected exit code to be non-zero, but got ${result.code}`);
-    })
-  );
+    let result = await runCommand(path.join('.', 'node_modules', 'ember-cli', 'bin', 'ember'), 'build');
+    expect(result.code).to.equal(0, `expected exit code to be zero, but got ${result.code}`);
 
-  it(
-    'ember build generates instrumentation files when viz is enabled',
-    co.wrap(function*() {
-      process.env.BROCCOLI_VIZ = '1';
+    // add something broken to the project to make build fail
+    fs.appendFileSync(appJsPath, '{(syntaxError>$@}{');
 
-      yield runCommand(path.join('.', 'node_modules', 'ember-cli', 'bin', 'ember'), 'build', {
-        env: {
-          BROCCOLI_VIZ: '1',
-        },
-      }).finally(() => {
-        delete process.env.BROCCOLI_VIZ;
-      });
+    result = await expect(runCommand(path.join('.', 'node_modules', 'ember-cli', 'bin', 'ember'), 'build')).to.be
+      .rejected;
 
-      [
-        'instrumentation.build.0.json',
-        'instrumentation.command.json',
-        'instrumentation.init.json',
-        'instrumentation.shutdown.json',
-      ].forEach(instrumentationFile => {
-        expect(fs.existsSync(instrumentationFile)).to.equal(true);
+    expect(result.code).to.not.equal(0, `expected exit code to be non-zero, but got ${result.code}`);
+  });
 
-        let json = fs.readJsonSync(instrumentationFile);
-        expect(Object.keys(json)).to.eql(['summary', 'nodes']);
+  it('ember build generates instrumentation files when viz is enabled', async function() {
+    process.env.BROCCOLI_VIZ = '1';
+
+    await runCommand(path.join('.', 'node_modules', 'ember-cli', 'bin', 'ember'), 'build', {
+      env: {
+        BROCCOLI_VIZ: '1',
+      },
+    });
+    delete process.env.BROCCOLI_VIZ;
```

Comment:
this should be in a try/finally block to match the previous behavior

Reply:
👍 Thanks, I updated all finally blocks.